### ISSUE TITLE
Feature request: ability to add graphql-middlewares

### ISSUE THREAD

**Issue by vincenzo** [graphile/postgraphile#917]
This is to request the ability to decorate the final executable schema with [graphql-middleware](https://github.com/prisma/graphql-middleware).

**Comment by benjie** [comment]
Nope; moving this back to graphile-engine - it's a generic concern - let's add a new `finalize` hook that's called once the schema is generated for any badly behaved libraries that mutate a previously generated schema.
